### DISCUSSION TITLE
Merge Conflict Resolver is missing

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#14033] [Q&A] [answered]
Hi All,
I'm trying to merge resolver working, I can see in Theia Cloud IDE Merge Conflict Resolver it is working. However, when it come to the local Theia, I'm unable to get it working. I compared both the package.json in browser, and plugins in Theia Blueprint and in my local one, still I'm unable to get it working.

It would be of great help if someone could help with the step involved in setting up Merge Conflict in Vanilla Theia locally. 

Here's a picture of merge conflict resolver plugin that I trying to get working,
![image](https://github.com/user-attachments/assets/cce054e6-bbda-4bf7-9025-4e284f787703)

Thanks in advance

**Comment by msujew** [comment]
Hey @maheshrajrp,

are you using the `@theia/git` package in your app? The package is in the process of being deprecated (see https://github.com/eclipse-theia/theia/issues/12745). The recommendation is to use the `vscode.git` extension.

**Reply by konstantin-tarasov-volantetech** [reply]
Right `@theia/git` is deprecated.

[vscode extensions](https://open-vsx.org/extension/vscode/git) are also doesn't provide **merge conflict resolver**.

I've tried [latest instruction](https://theia-ide.org/docs/composing_applications/#consuming-vs-code-extensions) locally and it comes without **merge conflict resolver**.

Running [theia cloud](https://try.theia-cloud.io/), you can see the **merge conflict resolver** (looks exactly like on the screenshot above).

installing [theia blueprint](https://github.com/eclipse-theia/theia-blueprint) locally, it comes without **merge conflict resolver**.

So the question, how to make it works.

**Reply by msujew** [reply]
@JonasHelming Do you know which extension provides the merge conflict resolver?

**Reply by konstantin-tarasov-volantetech** [reply]
I guess, it is [merge-conflict](https://open-vsx.org/extension/vscode/merge-conflict). But seems, it doesn't work as expected.

You can take a look to [latest instruction](https://theia-ide.org/docs/composing_applications/#consuming-vs-code-extensions) and install the theia. It comes with [vscode-builtin-extensions-pack](https://open-vsx.org/api/eclipse-theia/builtin-extension-pack/1.50.1/file/eclipse-theia.builtin-extension-pack-1.50.1.vsix) (which includes **merge-conflict**). But doesn't provide the mentioned UI (on the screenshot).

**Reply by tsmaeder** [reply, marked as answer]
@konstantin-tarasov-volantetech I thknk we've got the culprit: the version of the built-ins you're using is 1.50.1. But if you look at the theia (or blueprint) package.json, the version there is 1.88.1. The version refers to the version of VS Code the built-ins come from, not the Thea release version.

**Reply by maheshrajrp** [reply]
Hi @tsmaeder ,

Thanks. I have another question, I remember using this Merge Conflict Resolver in Theia early days like v1.38, currently I'm working on project that uses v1.38 which is missing the merge conflict resolver. What would be your suggestion for that ? 

Upgrading Theia is like longshot owing to React version upgrade and our current libraries doesn't support them.

**Reply by tsmaeder** [reply]
@maheshrajrp go back to 1.38 and check the version of the built-ins that were used at that time in the eclipse-theia/theia repo. But t.b.h, I would update to a newer version. And keep updating to every community release. It's easier to incrementally update than to fall back and then do a big update.
If you're looking for professional support with old versions of Theia, there are companies that offer such services like EclipseSource, TypeFox or my company Castle Ridge Software.

**Comment by JonasHelming** [comment]
@tsmaeder Could you retest this in your local IDE?

**Comment by tsmaeder** [comment]
Also see https://github.com/eclipse-theia/theia-website/pull/587
